src/dolphin/_show_versions.py

```diff
@@ -0,0 +1,104 @@
+"""Utility methods to print system info for debugging.
+
+Adapted from `rasterio.show_versions`,
+which was adapted from `sklearn.utils._show_versions`
+which was adapted from `pandas.show_versions`
+"""
+import importlib
+import platform
+import sys
+
+__all__ = ["show_versions"]
+
+
+def _get_sys_info():
+    """System information.
+
+    Returns
+    -------
+    dict:
+        system and Python version information
+    """
+    blob = [
+        ("python", sys.version.replace("\n", " ")),
+        ("executable", sys.executable),
+        ("machine", platform.platform()),
+    ]
+
+    return dict(blob)
+
+
+def _get_opera_info():
+    """Information on system on core modules.
+
+    Returns
+    -------
+    dict:
+        dolphin / opera module information
+    """
+    # pylint: disable=import-outside-toplevel
+    import dolphin
+
+    blob = [
+        ("dolphin", dolphin.__version__),
+        # optionals
+        ("isce3", _get_version("isce3")),
+        ("compass", _get_version("compass")),
+    ]
+
+    return dict(blob)
+
+
+def _get_version(modname):
+    try:
+        if modname in sys.modules:
+            mod = sys.modules[modname]
+        else:
+            mod = importlib.import_module(modname)
+        try:
+            return mod.__version__
+        except AttributeError:
+            return mod.version
+    except ImportError:
+        return None
```

Comment:
In general, I think it's a good idea to minimize the scope of your try/except blocks so that you're not accidentally catching more than you intended.

Let's rearrange this a bit so that only the line that might raise an `ImportError` is inside the `try: ...; except ImportError` scope.

```suggestion
    if modname in sys.modules:
        mod = sys.modules[modname]
    else:
        try:
            mod = importlib.import_module(modname)
        except ImportError:
            return None
    try:
        return mod.__version__
    except AttributeError:
        return mod.version
```